### DISCUSSION TITLE
shouldDirty not working correctly with Controller?

### DISCUSSION THREAD

**Discussion by ialsowalkdogs** [react-hook-form/react-hook-form#3650] [Q&A]
Hi,

I have two fields in my form that are dependent on each other, i.e. when one is changes, the other should retain default/current value. However that dos not seem to always happen predictably, is there something missing?

CodeSandbox: [https://codesandbox.io/s/fervent-roentgen-81c43](https://codesandbox.io/s/fervent-roentgen-81c43)
To reproduce, try changing the value of any of the Selects. Two thing happen:
* for Select 1, both fields will dirty on second change
* for Select 2, both fields will not dirty

**Comment by bluebill1049** [comment]
you are setting input with the same value, hook form will do a compare and not set dirty to `true`, if the value is not changed.

**Reply by ialsowalkdogs** [reply]
Thanks, that makes sense. Is there a way to forcibly dirty it for what I need?

**Reply by bluebill1049** [reply]
hmmm you will have to reset the `defaultValues` with reset API, or try to add a space? (hacky)

**Reply by ialsowalkdogs** [reply]
But then `reset` also does not dirty the fields? I rely on checking whether the field is dirty/has a default when processing the form data and I would like to keep that. Guess I will have to hack around it, thanks for your help!

**Reply by ialsowalkdogs** [reply]
Just FYI, I tried setting the field in question to an empty value and then to the correct one but that also does not dirty the field: https://codesandbox.io/s/fervent-roentgen-81c43?file=/src/App.tsx:1876-1908 I am guessing because the field will be dirtied only when a rerender is triggered?

**Reply by bluebill1049** [reply]
what's wrong the demo? yes it will need a rerender to update the dirty state
